network/p2p/gossip/mempool.go

```diff
@@ -0,0 +1,66 @@
+// Copyright (C) 2019-2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package gossip
+
+import (
+	"fmt"
+
+	"github.com/ava-labs/avalanchego/ids"
+
+	xmempool "github.com/ava-labs/avalanchego/vms/txs/mempool"
+)
+
+var ErrDuplicateTx = xmempool.ErrDuplicateTx
+
+type Mempool[T Gossipable] interface {
+	Add(tx T) error
```

Comment:
I don't mind us changing names to be `tx` instead of `gossipable` going forwards, but whatever we do we should be consistent in this diff (inconsistent w/ `handleIncomingGossipables`). I actually prefer `tx` naming, the only reason we use `gossipables` in the current code is because i thought we were going to use this for ip gossip, but we only ended up using this for txs.

Reply:
let me name the `tx` in the gossip package `gossipable` for now. We can make a change on the next PR. I want to keep the code consistent and make continues improvements.